lib/Controller/RoomController.php

```diff
@@ -1923,6 +1926,77 @@ public function verifyDialInPin(string $pin): DataResponse {
 		return new DataResponse($this->formatRoom($this->room, $participant, skipLastMessage: true));
 	}
 
+	/**
+	 * Direct dial-in (SIP bridge)
+	 *
+	 * @param string $phoneNumber Phone number that is called
+	 * @param string $caller Phone number of the person calling in
+	 * @return DataResponse<Http::STATUS_OK, TalkRoom, array{}>|DataResponse<Http::STATUS_UNAUTHORIZED|Http::STATUS_NOT_FOUND|Http::STATUS_NOT_IMPLEMENTED, null, array{}>
+	 *
+	 * 200: Participant returned
+	 * 401: SIP request invalid
+	 * 404: Participant not found
+	 * 501: SIP dial-in is not configured
+	 */
+	#[PublicPage]
+	#[BruteForceProtection(action: 'talkSipBridgeSecret')]
+	#[OpenAPI(scope: 'backend-sipbridge')]
+	#[RequireRoom]
+	public function directDialIn(string $phoneNumber, string $caller): DataResponse {
+		try {
+			if (!$this->validateSIPBridgeRequest($phoneNumber)) {
+				$response = new DataResponse(null, Http::STATUS_UNAUTHORIZED);
+				$response->throttle(['action' => 'talkSipBridgeSecret']);
+				return $response;
+			}
+		} catch (UnauthorizedException) {
+			$response = new DataResponse(null, Http::STATUS_UNAUTHORIZED);
+			$response->throttle(['action' => 'talkSipBridgeSecret']);
+			return $response;
+		}
+
+		if (!$this->talkConfig->isSIPConfigured()) {
+			return new DataResponse(null, Http::STATUS_NOT_IMPLEMENTED);
+		}
+
+		try {
+			$entity = $this->phoneService->getAccountToCallForPhoneNumber($phoneNumber);
+		} catch (DoesNotExistException) {
+			return new DataResponse(null, Http::STATUS_NOT_FOUND);
+		}
+
+		$room = null;
+		$cleanedCaller = $this->phoneNumberUtil->convertToStandardFormat($caller);
+		if ($cleanedCaller !== null) {
+			$objectId = $cleanedCaller;
+			try {
+				$room = $this->manager->getRoomForUserByObject($entity->getActorId(), Room::OBJECT_TYPE_PHONE, $objectId);
+				$this->logger->debug('Reusing existing direct dial-in conversation for caller ' . $cleanedCaller . ' and user ' . $entity->getActorId());
+			} catch (RoomNotFoundException) {
+			}
+		} else {
+			$objectId = Room::OBJECT_TYPE_PHONE;
+		}
+
+		if (!$room instanceof Room) {
+			$user = $this->userManager->get($entity->getActorId());
+			try {
+				$room = $this->roomService->createConversation(
+					Room::TYPE_GROUP,
```

Comment:
@fancycode can you give it a try if group is okay, or whether we need:
```suggestion
					Room::TYPE_PUBLIC,
```

Reply:
`TYPE_GROUP` works (I can hear the phone in the web and vice versa). Didn't test `TYPE_PUBLIC`.